### ISSUE TITLE
should not report error for `macro_rules`

### ISSUE THREAD

**Issue by bvanjoi** [rust-lang/rust-analyzer#15969]
**rust-analyzer version**: 0.3.1740

**rustc version**: rustc 1.76.0-nightly (3a85a5cfe 2023-11-20)

**relevant settings**: nothing

code:

```rs
macro_rules! macro_rules {
    () => {}
}

fn main() {
  macro_rules!{};
}
```

rust-analyzer indicates that there is a syntax error, however, rustc can compile the same code without any issues:

<img width="595" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/30187863/ab633f0f-e86a-4b9f-b89f-2cb154d6d0ac">

<img width="861" alt="image" src="https://github.com/rust-lang/rust-analyzer/assets/30187863/138d8e59-9a04-4cf6-a7fe-ad88282a08af">
